src/main/java/com/faforever/client/remote/gson/FactionTypeAdapter.java

```diff
@@ -27,6 +27,6 @@ public Faction read(JsonReader in) throws IOException {
       in.nextNull();
       return null;
     }
-    return Faction.fromString(in.nextString());
+    return Faction.fromFaValue(Integer.parseInt(in.nextString()));
```

Comment:
I prefer meaningful strings like "aeon" to be stored in the preferences rather than numbers, because of human readability (vs. magic numbers). Therefore, I suggest using `value.getString()` to write the value instead.